### DISCUSSION TITLE
KafkaConnector migration between clusters

### DISCUSSION THREAD

**Discussion by opsocomusr** [strimzi/strimzi-kafka-operator#5766] [Q&A] [answered]
We need to migrate our existing connectors(confluent) to strimzi's connect. Is there any configuration parameter where we can define the connector name similar to the name that is configured in the older cluster?

Currently, the name that is configured in the `.metadata.name` is considered as the connector name and we will not be able to configure names similar to the older cluster since it has uppercase characters. The problem we are facing is that if configure a different name in strimzi, it creates a new consumer group which is different from our previous cluster and the new connector starts picking up the data from the beginning.

If we have a configuration to define the connector name in the `KafkaConnector` custom resource, we will have the consumer group similar to our previous cluster which can be migrated to strimzi with MM2 and we will be able to perform a seamless migration of the connectors.

**Comment by scholzj** [comment, marked as answer]
The connector name is always the name of the `KafkaConnector` resource. There is no way to change it. So I guess you have two options => 
* Set the offsets manually for the new group to continue where the old connector left it
* Disable the Connector operator which will let you manage the connectors using the REST API directly and you can give them any names you want. But you will not be able to use the `KafkaConnector` resources.

**Reply by opsocomusr** [reply]
Thanks for the quick response!

If we disable the connector operator, is it still possible to use the external configuration providers (secrets, configmaps) with the REST API?

**Reply by scholzj** [reply]
Yes, that should be still possible. The configuration providers are running inside Kafka Connect and not in the operator. So you can still use them in the configuration when using the REST APi.
